app/dependencies.py

```diff
@@ -164,6 +164,13 @@ def get_redis_client(
     return redis_client
 
 
+def get_scheduler(settings: Settings = Depends(get_settings)) -> Scheduler:
+    global scheduler
+    if scheduler is None:
+        scheduler = Scheduler() if settings.REDIS_HOST != "" else DummyScheduler()
```

Comment:
Instead of using a `DummyScheduler`, we could add a check in `Scheduler` to verify if the redis host is defined. If this is not the case we can maybe drop tasks

Reply:
Yes but we would still need to call the methods with no implementation when sending notifications thus the dummy Scheduler.
Thonyk said it was ok

Reply:
This indeed works, but we never created a *dummy* class like this one in the rest of the code. We usually check if Redis/Firebase is configured, and if this is not the case, we just return without executing the rest of the method.

I would think that using this solution would be more coherent with the rest of our code base. Moreover, it only adds a few lines of code instead of defining a complete new class `DummyScheduler`

Reply:
This does work but is however not coherent with using dependency injection. Doing nothing seems more of a get around than injecting a custom mock implementation.
If consistency is an issue, it might be worthwhile considering to change the rest and not this change.